### PR TITLE
chore(repo): Add backport script

### PR DESCRIPTION
**Pull request by LekoArts**
## Description

This adds a `backport` script inside `scripts`. You'll be able to do:

```shell
cd scripts
node backport.mjs main-v4 1234
```

Example backport PR: https://github.com/clerkinc/javascript/pull/1958

If you have access to it, use the `CLERK_COOKIE_PAT` so that the bot user creates the PR.

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [x] (If applicable) [Documentation](https://github.com/clerkinc/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [ ] `@clerk/types`
- [ ] `@clerk/themes`
- [ ] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [x] `build/tooling/chore`

### CODE REVIEW

File: docs/PUBLISH.md [hunk 1]
```diff
@@ -42,3 +40,17 @@ Notes:
 ## Publishing canary package versions (`@canary`)
 
 We're still considering whether switching the repo into a `canary` mode for big, experimental features makes sense. There is no use case for this right now, however, we might experiment with the [changesets prerelease mode](https://github.com/changesets/changesets/blob/main/docs/prereleases.md) in the future.
+
+## Backporting PRs
```
Comment (LekoArts) [inline docs/PUBLISH.md:44]: Lemme know if you'd want any more details on this for that section

File: scripts/backport.mjs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    body: `Backporting #${pull_number} to the ${targetBranchName} branch\n\n(cherry picked from commit ${commitSha})`,
+  });
+
+  console.log(`\n---\n\nPR opened: ${pr.data.html_url}`);
+
+  await octokit.issues.addLabels({
+    owner,
+    repo,
+    issue_number: pr.data.number,
+    labels: [`cherry`],
```
Comment (LekoArts) [inline scripts/backport.mjs:89]: Short from of "cherry-picking"

**Comment by nikosdouvlis** [comment]
@LekoArts Good job automating this one :) Do you think we could turn this into a GH workflow using the `  workflow_dispatch` trigger so it can be manually triggered while also accepting inputs? 

Or maybe even going a bit further, another similar idea would be to create a special tag (eg `backport`) and have your script run after merge for every tagged PR

**Review by nikosdouvlis — COMMENTED** [review]
@LekoArts Good job automating this one :) Do you think we could turn this into a GH workflow using the   workflow_dispatch trigger so it can be manually triggered while also accepting inputs?

Or maybe even going a bit further, another similar idea would be to create a special tag (eg backport) and have your script run after merge for every tagged PR

**Comment by LekoArts** [comment]
The script in its current form wouldn't be ready for that as it doesn't handle any error cases / things besides the happy path. Once you have merge conflicts the script won't finish successfully so it's better to run this locally so that you can also resolve it locally directly.

**Review by nikosdouvlis — APPROVED** [review]

**Pull request merged**